src/main/java/com/google/devtools/build/lib/syntax/StringModule.java

```diff
@@ -909,9 +916,21 @@ public Integer count(String self, String sub, Integer start, Object end)
             defaultValue = "None",
             doc = "optional position at which to stop comparing.")
       })
-  public Boolean endsWith(String self, String sub, Integer start, Object end)
+  public Boolean endsWith(String self, Object sub, Integer start, Object end)
       throws ConversionException {
-    return pythonSubstring(self, start, end, "'end' operand of 'endswith'").endsWith(sub);
+    String str = pythonSubstring(self, start, end, "'end' operand of 'endswith'");
+    if (sub instanceof String) {
+      return str.endsWith((String)sub);
+    }
+
+    @SuppressWarnings("unchecked")
+    SkylarkList<String> subs = (SkylarkList<String>) sub;
+    for (String s : subs.getImmutableList()) {
```

Comment:
Why are we allocating a copy of the list here, only to throw it away?

Reply:
Done